Sources/Logging/Destinations/Log+ConsoleLogDestination.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  Log+ConsoleLogDestination.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 07/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public extension Log {
+
+    public enum ConsoleLogDestinationError: Swift.Error {
+        case itemFormatFailed(Log.Item, Swift.Error)
+    }
+
+    public class ConsoleLogDestination<ItemFormatter: LogItemFormatter>: LogDestination
+    where ItemFormatter.Output == String {
+
+        public typealias OutputClosure = ((Level, String) -> Void)
+        public typealias LogMetadataClosure = ([AnyHashable : Any]) -> (Level, String)
+
+        public let formatter: ItemFormatter
+
+        public let minLevel: Level
+
+        private let output: OutputClosure
+        private let logMetadata: LogMetadataClosure?
+
+        // MARK: - Lifecycle
+
+        public init(formatter: ItemFormatter,
+                    minLevel: Level = .error,
+                    output: @escaping OutputClosure = { (level, message) in print(message) },
+                    logMetadata: LogMetadataClosure? = nil) {
```

Comment:
Isn't `metadata` enough?

Reply:
`metadata` might be interpreted as the metadata itself, whereas the objective is for this to be the closure that "logs the metadata"

Reply:
Agree with you, although I think `log metadata` reads as a verb instead of a callback. If I read the signature, and see `logMetadata`, my first reaction is to assume it expects a boolean. But on the other hand, I agree it should be named purely `metadata`

I can't come up with a better naming though. Obviously, this is not a blocker; just me being picky. We have the `output` closure. Could we call this one `metadataOutput`?

Reply:
Not a very big fan of `metadataOutput`, since it doesn't output the metadata directly. It creates the input (level and message) for the `output` closure to log 😝 

Perhaps `metadataMessage`? If not, we can just leave it as is for now, I guess 😇 

Meanwhile, I'll add documentation to the API's so (hopefully) it should be clearer to the users.